sdk/core/azure-core/inc/azure/core/credentials.hpp

```diff
@@ -47,12 +47,13 @@ namespace Azure { namespace Core {
     /**
      * @brief Get an authentication token.
      *
-     * @param context #Azure::Core::Context so that operation can be cancelled.
      * @param tokenRequestOptions Options to get the token.
+     * @param context #Azure::Core::Context so that operation can be cancelled.
+     *
      */
     virtual AccessToken GetToken(
-        Context const& context,
-        Http::TokenRequestOptions const& tokenRequestOptions) const = 0;
+        Http::TokenRequestOptions const& tokenRequestOptions,
+        Context const& context) const = 0;
```

Comment:
I am not sure about this one, since it is being moved after options. Should these options be optional and defaulted?

Reply:
cc @antkmsft

Reply:
@JeffreyRichter - let me know what you think here around `TokenCredential::GetToken`
```C++
TokenCredential::GetToken(Http::TokenRequestOptions const& tokenRequestOptions, Azure::Core::Context const& context)
```

Reply:
@schaabs and @antkmsft - can you please take a look at this particular change. If we make `TokenRequestOptions` optional, like most options, context would move to be the first parameter.

Reply:
There can be no default TokenRequestOptions, for the reason they all look different depending on the client, and no empty value would make any sense, you are writing an erroneous code if you put empty scopes here. This was discussed in https://github.com/Azure/azure-sdk-for-cpp/pull/1527#discussion_r568968684

Let's rename the struct from `*Options` to `TokenRequestContext` or anything of that sort - it seems the problem truly is between the name suffix and optionality; But let's not make this parameter to have default value, a bad and problematic one, to follow the pattern.

Reply:
Alright, I will leave it as I have it then for now.

Though unrelated to this PR, I would love to hear Scott's thoughts on that conversation you linked to Anton. What do other languages do here?

> Let's rename the struct

Can you please file an issue and we can settle on a name. `TokenRequestContext` is problematic given we have a Context type already and that's the other parameter being passed in here.